src/lib/operators/abstract_aggregate_operator.hpp

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include "expression/aggregate_expression.hpp"
+#include "operators/abstract_operator.hpp"
+#include "operators/abstract_read_only_operator.hpp"
+#include "type_comparison.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * Aggregates are defined by the column (ColumnID for Operators, LQPColumnReference in LQP) they operate on and the aggregate
+ * function they use. COUNT() is the exception that does not necessarily use a column, which is why column is optional.
+ *   COUNT(*) does not use a column and returns the number of rows
+ *   COUNT(<column>) does use a column and returns the number of rows with non-null values in <column>
+ * Optionally, an alias can be specified to use as the output name. //TODO(anyone) copied from hash aggregate, but does this really happen here? Isnt that rather the task of a further downstream projection?
+ *
+ * Further, the aggregate operator is used to perfom distinct operations. This functionality is achieved by having no aggregates.
```

Comment:
I would capitalize DISTINCT to make a difference to the adjective. Btw, are there multiple DISTINCT operations? Or is it just a single one?